rhc-ose-ansible/roles/secure-registry/tasks/main.yaml

```diff
@@ -0,0 +1,164 @@
+
+  - name: "Adding secrets to registry's service account"
+    command: "{{openshift.common.client_binary}} secrets add serviceaccounts/{{registry_serviceaccount}} secrets/{{registry_secret_name}} -n {{openshift_registry_project}}"
+    run_once: true
+    delegate_to: "{{ groups['masters'][0] }}"
+
+  - name: "Adding secrets to default account"
+    command: "{{openshift.common.client_binary}} secrets add serviceaccounts/default secrets/{{registry_secret_name}} -n {{openshift_registry_project}}"
+    run_once: true
+    delegate_to: "{{ groups['masters'][0] }}"
+
+  - name: "Ceanup any previous runs"
+    shell: "{{openshift.common.client_binary}} get deploymentConfig/{{registry_dc}} -o yaml | grep {{registry_secret_name}}"
+    run_once: true
+    delegate_to: "{{ groups['masters'][0] }}"
+    register: dc_cleanup
+    failed_when: dc_cleanup.rc != 0  and dc_cleanup.rc != 1
+
+  - name: "Add secret volumes to registry deploymentConfiguration"
+    command: "{{openshift.common.client_binary}} volume deploymentConfig/{{registry_dc}} --add --type=secret --secret-name={{registry_secret_name}} -m /etc/secrets --overwrite"
+    run_once: true
+    delegate_to: "{{ groups['masters'][0] }}"
+    when: dc_cleanup.rc == 0
+    
+  - name: "Add secret volumes to registry deploymentConfiguration"
+    command: "{{openshift.common.client_binary}} volume deploymentConfig/{{registry_dc}} --add --type=secret --secret-name={{registry_secret_name}} -m /etc/secrets"
+    run_once: true
+    delegate_to: "{{ groups['masters'][0] }}"
+    when: dc_cleanup.rc == 1
+
+  - name: "Enable TLS"
+    command: "{{openshift.common.client_binary}} env deploymentConfig/{{registry_dc}} REGISTRY_HTTP_TLS_CERTIFICATE={{registry_certificate}} REGISTRY_HTTP_TLS_KEY={{registry_key}} --overwrite"
+    run_once: true
+    delegate_to: "{{ groups['masters'][0] }}"
+
+  - name: "Updating liveness probe to HTTPS scheme"
+    command: "{{openshift.common.client_binary}} patch deploymentConfig/{{registry_dc}} --api-version=v1 -p '{{liveness_patch}}'"
+    run_once: true
+    delegate_to: "{{ groups['masters'][0] }}"
+
+  - name: "Updating readinessProbe to HTTPS scheme"
+    command: "{{openshift.common.client_binary}} patch deploymentConfig/{{registry_dc}} --api-version=v1 -p '{{readiness_patch}}'"
+    run_once: true
+    delegate_to: "{{ groups['masters'][0] }}"
+
+  - name: "Create registry certificates directory for serviceIP"
+    file: "state=directory path=/etc/docker/certs.d/{{registry_svc_ip}}:{{item}} mode=755"
+    with_items: "{{registry_svc_ports}}"
+
+  - name: "Create registry certificates directory for kubernetes service"
+    file: "state=directory path=/etc/docker/certs.d/{{registry_dc}}.default.svc.cluster.local:{{item}} mode=755"
+    with_items: "{{registry_svc_ports}}"
+
+  - name: "Copy registry cert to kubernetes service directory"
+    command: "scp {{registry_ca}} {{item[0]}}:/etc/docker/certs.d/{{registry_dc}}.default.svc.cluster.local:{{item[1]}}"
```

Comment:
@JayKayy this line appears to assume that ansible is run from the master, which is often not the case. When I try to run ansible from some remote host, then this task just hangs as the scp line doesn't make sense. 

@abutcher @detiberus what's the right way to handle syncing certs around?